src/Compilers/Core/CodeAnalysisTest/Collections/EnumerableExtensionsTests.cs

```diff
@@ -22,5 +25,54 @@ public void AsSingleton()
             Assert.Equal(1, Enumerable.Range(1, 1).AsSingleton());
             Assert.Equal(0, Enumerable.Range(1, 2).AsSingleton());
         }
+
+        private class ReadOnlyList<T> : IReadOnlyList<T>
+        {
+            private readonly T[] _items;
+
+            public ReadOnlyList(params T[] items)
+            {
+                _items = items;
+            }
+
+            public T this[int index] => _items[index];
+            public int Count => _items.Length;
+            public IEnumerator<T> GetEnumerator() => throw new NotImplementedException();
+            IEnumerator IEnumerable.GetEnumerator() => throw new NotImplementedException();
+        }
+
+        private class SignlessEqualityComparer : IEqualityComparer<int>
+        {
+            public bool Equals(int x, int y) => Math.Abs(x) == Math.Abs(y);
+            public int GetHashCode(int obj) => obj.GetHashCode();
```

Comment:
>GetHashCode [](start = 51, length = 11)

This looks wrong.

Reply:
Better to throw an exception if not expected to be used.

---
In reply to: [394046102](https://github.com/dotnet/roslyn/pull/42451#discussion_r394046102) [](ancestors = 394046102)